Symfony5 support

nanasess opened the pull request:
<!-- 以下を参考にコメントを作成してください。 -->

## 概要(Overview・Refs Issue)
<!-- PullRequestの目的、関連するIssue番号など -->
- Symfony5 へ composer update を実行し、影響範囲を調査する
- その他の依存パッケージは `php: "^7.4 || ^8.1"` でサポートされる最新バージョンに可能な限りアップデートする
- とり急ぎ、PHP7.4 及び PHP8.1 で TOP ページが表示されるところまで対応
- `FIXME` が付与されているコミットログは、とり急ぎエラーが出ないようコメントアウト等の措置をしただけなので、修正が必要
- #5188
   - doctrine/persistence のバージョンアップに伴い、 Repository クラスで使用してる RegistryInterface の namespace 変更が必要
   - doctrine/migrations のバージョンアップに伴い、マイグレーション関連のコードの大幅な修正が必要
   - bheller/images-generator は PHP8 に対応していないため削除。 eccube:fixtures:generate コマンドの修正が必要
   - fzaninotto/faker が廃止。fakerphp/faker を利用するようになったため、テスト関連のコードの修正が必要かも- 

### 削除されるパッケージ
- symfony/security
- symfony/web-server-bundle
- symfony/debug
- suncat/mobile-detect-bundle
- pimple/pimple

## メジャーバージョンアップするパッケージ
- knplabs/knp-paginator-bundle
- nesbot/carbon
- phpunit/phpunit
- vlucas/phpdotenv
- sensio/framework-extra-bundle
- monolog/monolog
- guzzlehttp/guzzle
- friendsofphp/php-cs-fixer
- doctrine/inflector
- doctrine/cache
- doctrine/persistence
- doctrine/dbal
- doctrine/orm

上記の他、 swiftmailer を symfony/mailer に変更する必要がある

## その他の懸念事項

- **twig/extensions が開発終了している**
   - Twig Core Extra extensions に移行した
   - [IntlExtension](https://github.com/EC-CUBE/ec-cube/blob/4.1/src/Eccube/Twig/Extension/IntlExtension.php) の変更が必要
- twig/twig "^3.3" に更新するかどうか
   - プラグインやサンプルコードで使用している非推奨コードが無いかどうか精査 https://twig.symfony.com/doc/2.x/deprecated.html
   - `Twig_*` の class alias が廃止されるため、プラグインなどで使用している場合は変更が必要
- doctrine/orm が friendsofphp/proxy-manager-lts 経由で依存している laminas/laminas-code のサポート期間が短く、 PHP8.2 がサポートされるか心配

## 方針(Policy)
<!-- このPullRequestを作るにあたって考慮したものや除外した内容
  - 例）Symfony のXXにならって作成、3系で同様の仕様があったため など -->

## 実装に関する補足(Appendix)
<!-- コードだけではわかりづらい点など、実装するにあたってレビューアに追加で伝えておきたいこと -->

<details>

  - Removing symfony/web-server-bundle (v4.4.27)
  - Removing symfony/security (v4.4.36)
  - Removing symfony/polyfill-php70 (v1.20.0)
  - Removing symfony/mime (v5.4.2)
  - Removing symfony/http-client-contracts (v2.5.0)
  - Removing symfony/debug (v4.4.36)
  - Removing suncat/mobile-detect-bundle (v1.1.1)
  - Removing pimple/pimple (v1.1.1)
  - Removing phpunit/php-token-stream (3.1.3)
  - Removing laminas/laminas-zendframework-bridge (1.4.1)
  - Removing laminas/laminas-eventmanager (3.4.0)
  - Removing kylekatarnls/update-helper (1.2.1)
  - Removing jdorn/sql-formatter (v1.2.17)
  - Removing fzaninotto/faker (v1.9.2)
  - Removing doctrine/reflection (1.2.2)
  - Removing doctrine/doctrine-cache-bundle (1.4.0)
  - Removing composer/package-versions-deprecated (1.11.99.4)
  - Removing bheller/images-generator (1.0.1)
  - Upgrading symfony/flex (v1.17.6 => v1.18.5): Extracting archive
  - Upgrading symfony/deprecation-contracts (v2.5.0 => v2.5.1): Extracting archive
  - Installing psr/http-client (1.0.1): Extracting archive
  - Installing psr/http-factory (1.0.1): Extracting archive
  - Upgrading guzzlehttp/psr7 (1.8.5 => 2.2.1): Extracting archive
  - Upgrading guzzlehttp/guzzle (6.5.5 => 7.4.2): Extracting archive
  - Upgrading sebastian/version (2.0.1 => 3.0.2): Extracting archive
  - Installing sebastian/type (3.0.0): Extracting archive
  - Upgrading sebastian/resource-operations (2.0.2 => 3.0.3): Extracting archive
  - Upgrading sebastian/recursion-context (3.0.1 => 4.0.4): Extracting archive
  - Upgrading sebastian/object-reflector (1.1.2 => 2.0.4): Extracting archive
  - Upgrading sebastian/object-enumerator (3.0.4 => 4.0.4): Extracting archive
  - Upgrading sebastian/global-state (2.0.0 => 5.0.5): Extracting archive
  - Upgrading sebastian/exporter (3.1.4 => 4.0.4): Extracting archive
  - Upgrading sebastian/environment (4.2.4 => 5.1.4): Extracting archive
  - Upgrading sebastian/diff (3.0.3 => 4.0.4): Extracting archive
  - Upgrading sebastian/comparator (3.0.3 => 4.0.6): Extracting archive
  - Installing sebastian/code-unit (1.0.8): Extracting archive
  - Installing sebastian/cli-parser (1.0.1): Extracting archive
  - Upgrading phpunit/php-timer (2.1.3 => 5.0.3): Extracting archive
  - Upgrading phpunit/php-text-template (1.2.1 => 2.0.4): Extracting archive
  - Installing phpunit/php-invoker (3.1.1): Extracting archive
  - Upgrading phpunit/php-file-iterator (2.0.5 => 3.0.6): Extracting archive
  - Installing sebastian/lines-of-code (1.0.3): Extracting archive
  - Installing sebastian/complexity (2.0.2): Extracting archive
  - Upgrading sebastian/code-unit-reverse-lookup (1.0.2 => 2.0.3): Extracting archive
  - Upgrading phpunit/php-code-coverage (6.1.4 => 9.2.15): Extracting archive
  - Upgrading symfony/polyfill-ctype (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading phpdocumentor/type-resolver (1.6.0 => 1.6.1): Extracting archive
  - Upgrading doctrine/instantiator (1.4.0 => 1.4.1): Extracting archive
  - Upgrading phar-io/version (2.0.1 => 3.2.1): Extracting archive
  - Upgrading phar-io/manifest (1.0.3 => 2.0.3): Extracting archive
  - Upgrading myclabs/deep-copy (1.10.2 => 1.11.0): Extracting archive
  - Upgrading symfony/polyfill-mbstring (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading phpunit/phpunit (7.5.20 => 9.5.20): Extracting archive
  - Upgrading codeception/phpunit-wrapper (7.8.2 => 9.0.7): Extracting archive
  - Upgrading symfony/yaml (v4.4.36 => v5.4.3): Extracting archive
  - Upgrading symfony/polyfill-php80 (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading symfony/finder (v4.4.36 => v5.4.3): Extracting archive
  - Installing psr/event-dispatcher (1.0.0): Extracting archive
  - Upgrading symfony/event-dispatcher-contracts (v1.1.11 => v2.5.1): Extracting archive
  - Upgrading symfony/event-dispatcher (v4.4.34 => v5.4.3): Extracting archive
  - Upgrading symfony/css-selector (v4.4.27 => v5.4.3): Extracting archive
  - Upgrading symfony/polyfill-intl-normalizer (v1.24.0 => v1.25.0): Extracting archive
  - Installing symfony/polyfill-intl-grapheme (v1.25.0): Extracting archive
  - Installing symfony/string (v5.4.3): Extracting archive
  - Upgrading psr/container (1.1.1 => 1.1.2): Extracting archive
  - Upgrading symfony/service-contracts (v2.5.0 => v2.5.1): Extracting archive
  - Upgrading symfony/polyfill-php73 (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading symfony/console (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading codeception/stub (3.0.0 => 4.0.2): Extracting archive
  - Upgrading codeception/codeception (4.1.28 => 4.1.31): Extracting archive
  - Upgrading symfony/dom-crawler (v4.4.36 => v5.4.6): Extracting archive
  - Upgrading symfony/browser-kit (v4.4.27 => v5.4.3): Extracting archive
  - Upgrading justinrainbow/json-schema (5.2.11 => 5.2.12): Extracting archive
  - Upgrading symfony/process (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/filesystem (v4.4.27 => v5.4.7): Extracting archive
  - Upgrading seld/jsonlint (1.8.3 => 1.9.0): Extracting archive
  - Upgrading react/promise (v2.8.0 => v2.9.0): Extracting archive
  - Upgrading composer/pcre (1.0.0 => 3.0.0): Extracting archive
  - Upgrading composer/xdebug-handler (2.0.4 => 3.0.3): Extracting archive
  - Upgrading composer/semver (3.2.7 => 3.3.2): Extracting archive
  - Upgrading composer/composer (2.2.4 => 2.3.5): Extracting archive
  - Upgrading symfony/routing (v4.4.34 => v5.4.3): Extracting archive
  - Upgrading symfony/polyfill-php81 (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading symfony/http-foundation (v4.4.36 => v5.4.6): Extracting archive
  - Upgrading symfony/var-dumper (v4.4.36 => v5.4.6): Extracting archive
  - Upgrading symfony/error-handler (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/http-kernel (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/dependency-injection (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/config (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/var-exporter (v5.4.2 => v5.4.7): Extracting archive
  - Upgrading symfony/cache-contracts (v2.5.0 => v2.5.1): Extracting archive
  - Upgrading symfony/cache (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/framework-bundle (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading doctrine/cache (1.12.1 => 2.1.1): Extracting archive
  - Upgrading doctrine/persistence (1.3.8 => 2.5.0): Extracting archive
  - Upgrading symfony/doctrine-bridge (v4.4.36 => v5.4.7): Extracting archive
  - Installing doctrine/sql-formatter (1.1.2): Extracting archive
  - Upgrading doctrine/dbal (2.13.7 => 3.3.5): Extracting archive
  - Upgrading doctrine/lexer (1.2.2 => 1.2.3): Extracting archive
  - Upgrading doctrine/doctrine-bundle (1.12.13 => 2.6.2): Extracting archive
  - Upgrading dama/doctrine-test-bundle (v5.0.5 => v6.7.5): Extracting archive
  - Upgrading symfony/polyfill-php72 (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading doctrine/inflector (1.3.1 => 2.0.4): Extracting archive
  - Upgrading doctrine/common (2.13.3 => 3.2.2): Extracting archive
  - Upgrading doctrine/orm (2.7.5 => 2.11.2): Extracting archive
  - Upgrading doctrine/data-fixtures (1.5.1 => 1.5.2): Extracting archive
  - Upgrading symfony/stopwatch (v4.4.27 => v5.4.5): Extracting archive
  - Upgrading laminas/laminas-code (3.4.1 => 4.5.1): Extracting archive
  - Upgrading friendsofphp/proxy-manager-lts (v1.0.5 => v1.0.7): Extracting archive
  - Upgrading doctrine/migrations (v1.8.1 => 3.5.0): Extracting archive
  - Upgrading doctrine/doctrine-migrations-bundle (v1.3.2 => 3.2.2): Extracting archive
  - Upgrading monolog/monolog (1.26.1 => 2.5.0): Extracting archive
  - Upgrading symfony/polyfill-intl-idn (v1.24.0 => v1.25.0): Extracting archive
  - Installing fakerphp/faker (v1.19.0): Extracting archive
  - Upgrading symfony/options-resolver (v4.4.30 => v5.4.3): Extracting archive
  - Upgrading php-cs-fixer/diff (v1.3.1 => v2.0.2): Extracting archive
  - Upgrading friendsofphp/php-cs-fixer (v2.19.3 => v3.8.0): Extracting archive
  - Upgrading twig/twig (v2.14.11 => v2.14.13): Extracting archive
  - Upgrading symfony/translation-contracts (v2.5.0 => v2.5.1): Extracting archive
  - Upgrading symfony/translation (v4.4.34 => v5.4.7): Extracting archive
  - Upgrading knplabs/knp-components (v1.3.10 => v3.5.0): Extracting archive
  - Upgrading knplabs/knp-paginator-bundle (v2.8.0 => v5.8.0): Extracting archive
  - Upgrading mobiledetect/mobiledetectlib (2.8.37 => 2.8.39): Extracting archive
  - Upgrading nesbot/carbon (1.39.1 => 2.57.0): Extracting archive
  - Upgrading robthree/twofactorauth (1.8.1 => 1.8.2): Extracting archive
  - Upgrading sensio/framework-extra-bundle (v5.6.1 => v6.2.6): Extracting archive
  - Upgrading symfony/asset (v4.4.27 => v5.4.7): Extracting archive
  - Upgrading symfony/twig-bridge (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/debug-bundle (v4.4.36 => v5.4.3): Extracting archive
  - Upgrading symfony/dotenv (v4.4.36 => v5.4.5): Extracting archive
  - Upgrading symfony/expression-language (v4.4.34 => v5.4.7): Extracting archive
  - Installing symfony/property-info (v5.4.7): Extracting archive
  - Upgrading symfony/property-access (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/polyfill-intl-icu (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading symfony/form (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/inflector (v4.4.34 => v5.4.3): Extracting archive
  - Upgrading symfony/intl (v4.4.36 => v5.4.5): Extracting archive
  - Upgrading symfony/maker-bundle (v1.36.4 => v1.38.0): Extracting archive
  - Upgrading symfony/monolog-bridge (v4.4.27 => v5.4.3): Extracting archive
  - Upgrading symfony/phpunit-bridge (v4.4.34 => v5.4.7): Extracting archive
  - Upgrading symfony/polyfill-iconv (v1.24.0 => v1.25.0): Extracting archive
  - Upgrading symfony/proxy-manager-bridge (v4.4.36 => v5.4.6): Extracting archive
  - Installing symfony/password-hasher (v5.4.3): Extracting archive
  - Installing symfony/security-core (v5.4.7): Extracting archive
  - Installing symfony/security-http (v5.4.5): Extracting archive
  - Installing symfony/security-guard (v5.4.3): Extracting archive
  - Installing symfony/security-csrf (v5.4.3): Extracting archive
  - Upgrading symfony/security-bundle (v4.4.36 => v5.4.5): Extracting archive
  - Upgrading symfony/serializer (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/swiftmailer-bundle (v3.5.3 => v3.5.4): Extracting archive
  - Upgrading symfony/templating (v4.4.36 => v5.4.3): Extracting archive
  - Upgrading symfony/validator (v4.4.36 => v5.4.7): Extracting archive
  - Upgrading symfony/twig-bundle (v4.4.36 => v5.4.3): Extracting archive
  - Upgrading symfony/web-profiler-bundle (v4.4.31 => v5.4.6): Extracting archive
  - Upgrading symfony/workflow (v4.4.27 => v5.4.3): Extracting archive
  - Installing phpoption/phpoption (1.8.1): Extracting archive
  - Installing graham-campbell/result-type (v1.0.4): Extracting archive
  - Upgrading vlucas/phpdotenv (v2.4.0 => v5.4.0): Extracting archive
  
</details>

## テスト（Test)
<!-- テストを行っている範囲など、レビューアが安心できるような情報 -->
### TODO
- [x] `ReflectionException: Class Eccube\Twig\Extension\IgnoreRoutingNotFoundExtension does not exist`
- [x] `Error: Call to undefined method Doctrine\Bundle\DoctrineBundle\Mapping\MappingDriver::addDriver()`
- [x] `TypeError: Argument 1 passed to Symfony\Component\Serializer\Encoder\XmlEncoder::__construct() must be of the type array, string given`
- [x] `An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1 near "FROM": syntax error`
- [x] `ArgumentCountError: Too few arguments to function Twig\Environment::loadTemplate()`
- [x] `LogicException: Booting the kernel before calling "Symfony\Bundle\FrameworkBundle\Test\WebTestCase::createClient()" is not supported, the kernel should only be booted once.`
- [x] `TypeError: Argument 5 passed to Symfony\Component\HttpFoundation\File\UploadedFile::__construct() must be of the type bool, null given`
- [x] `Error: Call to a member function authenticate() on null`
   - https://github.com/EC-CUBE/ec-cube/pull/5225 を取り込んで対応
- [x] `TypeError: Argument 1 passed to Symfony\Bridge\Doctrine\Form\ChoiceList\IdReader::getIdValue() must be an object or null, int given`
- [x] OrderStateMachine の仕様変更
- [x] SameSiteCookie テストケースの修正
- [x] SwiftMailer から SymfonyMailer への変更
- [x] その他 WebTestCase の修正
- [x] Codeception の修正
- [x] Doctrine Migrations の仕様変更
- [x] env() の仕様変更
- [x] ContainerAwareCommand の廃止
- [x] PasswordEncoderInterface の仕様変更
- [x] Carbon のバージョンアップ
- [x] EC-CUBE本体で非推奨にしているメソッドの廃止

## 相談（Discussion）
<!-- 相談したいことや意見を求めたいこと -->

## マイナーバージョン互換性保持のための制限事項チェックリスト
<!-- マイナーバージョンでは、機能・プラグイン・デザインテンプレート互換性を損なう変更は原則取り込みません。 -->

- [ ] 既存機能の仕様変更はありません
- [ ] フックポイントの呼び出しタイミングの変更はありません
- [ ] フックポイントのパラメータの削除・データ型の変更はありません
- [ ] twigファイルに渡しているパラメータの削除・データ型の変更はありません
- [ ] Serviceクラスの公開関数の、引数の削除・データ型の変更はありません
- [ ] 入出力ファイル(CSVなど)のフォーマット変更はありません

## レビュワー確認項目

- [ ] 動作確認
- [ ] コードレビュー
- [ ] E2E/Unit テスト確認(テストの追加・変更が必要かどうか)
- [ ] 互換性が保持されているか
- [ ] セキュリティ上の問題がないか
  - [ ] 権限を超えた操作が可能にならないか
  - [ ] 不要なファイルアップロードがないか
  - [ ] 外部へ公開されるファイルや機能の追加ではないか
  - [ ] テンプレートでのエスケープ漏れがないか

nanasess committed:
Apply composer update

  - Removing bheller/images-generator (1.0.1)
  - Removing doctrine/doctrine-cache-bundle (1.4.0)
  - Removing doctrine/reflection (1.2.2)
  - Removing fzaninotto/faker (v1.9.2)
  - Removing jdorn/sql-formatter (v1.2.17)
  - Removing laminas/laminas-eventmanager (3.2.1)
  - Removing laminas/laminas-zendframework-bridge (1.1.1)
  - Removing phpunit/php-token-stream (3.1.2)
  - Removing symfony/debug-pack (v1.0.9)
  - Removing symfony/orm-pack (v1.0.8)
  - Removing symfony/profiler-pack (v1.0.5)
  - Upgrading behat/gherkin (v4.7.3 => v4.8.0)
  - Upgrading codeception/codeception (4.1.21 => 4.1.22)
  - Upgrading codeception/lib-innerbrowser (1.5.0 => 1.5.1)
  - Upgrading codeception/module-webdriver (1.2.1 => 1.4.0)
  - Upgrading codeception/phpunit-wrapper (7.8.2 => 9.0.6)
  - Upgrading codeception/stub (3.0.0 => 3.7.0)
  - Upgrading composer/ca-bundle (1.2.10 => 1.2.11)
  - Upgrading composer/composer (2.1.3 => 2.1.8)
  - Upgrading composer/package-versions-deprecated (1.11.99.2 => 1.11.99.4)
  - Upgrading composer/xdebug-handler (2.0.1 => 2.0.2)
  - Upgrading dama/doctrine-test-bundle (v5.0.5 => v6.6.0)
  - Upgrading doctrine/annotations (1.13.1 => 1.13.2)
  - Upgrading doctrine/collections (1.6.7 => 1.6.8)
  - Upgrading doctrine/common (2.13.3 => 3.1.2)
  - Upgrading doctrine/data-fixtures (1.3.3 => 1.5.1)
  - Upgrading doctrine/dbal (2.13.2 => 2.13.4)
  - Upgrading doctrine/doctrine-bundle (1.12.13 => 2.3.2)
  - Upgrading doctrine/doctrine-migrations-bundle (v1.3.2 => 3.1.1)
  - Upgrading doctrine/inflector (1.3.1 => 1.4.4)
  - Upgrading doctrine/lexer (1.0.2 => 1.2.1)
  - Upgrading doctrine/migrations (v1.8.1 => 3.2.1)
  - Upgrading doctrine/orm (2.7.5 => 2.8.5)
  - Upgrading doctrine/persistence (1.3.8 => 2.2.2)
  - Locking doctrine/sql-formatter (1.1.1)
  - Locking fakerphp/faker (v1.16.0)
  - Upgrading friendsofphp/php-cs-fixer (v2.19.0 => v2.19.2)
  - Upgrading justinrainbow/json-schema (5.2.10 => 5.2.11)
  - Upgrading laminas/laminas-code (3.4.1 => 4.4.3)
  - Upgrading mikey179/vfsstream (v1.6.9 => v1.6.10)
  - Upgrading nikic/php-parser (v4.11.0 => v4.13.0)
  - Upgrading phar-io/manifest (1.0.3 => 2.0.3)
  - Upgrading phar-io/version (2.0.1 => 3.1.0)
  - Upgrading phpdocumentor/reflection-common (2.1.0 => 2.2.0)
  - Upgrading phpdocumentor/reflection-docblock (4.3.4 => 5.2.2)
  - Upgrading phpdocumentor/type-resolver (1.0.1 => 1.5.1)
  - Upgrading phpspec/prophecy (v1.10.3 => 1.14.0)
  - Upgrading phpstan/phpstan (0.12.92 => 0.12.99)
  - Upgrading phpunit/php-code-coverage (6.1.4 => 9.2.7)
  - Upgrading phpunit/php-file-iterator (2.0.4 => 3.0.5)
  - Locking phpunit/php-invoker (3.1.1)
  - Upgrading phpunit/php-text-template (1.2.1 => 2.0.4)
  - Upgrading phpunit/php-timer (2.1.3 => 5.0.3)
  - Upgrading phpunit/phpunit (7.5.20 => 9.5.10)
  - Upgrading psr/container (1.0.0 => 1.1.1)
  - Locking sebastian/cli-parser (1.0.1)
  - Locking sebastian/code-unit (1.0.8)
  - Upgrading sebastian/code-unit-reverse-lookup (1.0.2 => 2.0.3)
  - Upgrading sebastian/comparator (3.0.3 => 4.0.6)
  - Locking sebastian/complexity (2.0.2)
  - Upgrading sebastian/diff (3.0.3 => 4.0.4)
  - Upgrading sebastian/environment (4.2.4 => 5.1.3)
  - Upgrading sebastian/exporter (3.1.3 => 4.0.3)
  - Upgrading sebastian/global-state (2.0.0 => 5.0.3)
  - Locking sebastian/lines-of-code (1.0.3)
  - Upgrading sebastian/object-enumerator (3.0.4 => 4.0.4)
  - Upgrading sebastian/object-reflector (1.1.2 => 2.0.4)
  - Upgrading sebastian/recursion-context (3.0.1 => 4.0.4)
  - Upgrading sebastian/resource-operations (2.0.2 => 3.0.3)
  - Locking sebastian/type (2.3.4)
  - Upgrading sebastian/version (2.0.1 => 3.0.2)
  - Upgrading seld/phar-utils (1.1.1 => 1.1.2)
  - Upgrading sensio/framework-extra-bundle (v5.5.7 => v5.6.1)
  - Upgrading symfony/asset (v4.4.25 => v4.4.27)
  - Upgrading symfony/browser-kit (v4.4.25 => v4.4.27)
  - Upgrading symfony/cache (v4.4.26 => v4.4.31)
  - Upgrading symfony/cache-contracts (v1.1.10 => v2.4.0)
  - Upgrading symfony/config (v4.4.26 => v4.4.30)
  - Upgrading symfony/console (v4.4.26 => v4.4.30)
  - Upgrading symfony/css-selector (v4.4.25 => v4.4.27)
  - Upgrading symfony/debug (v4.4.25 => v4.4.31)
  - Upgrading symfony/debug-bundle (v4.4.20 => v4.4.27)
  - Upgrading symfony/dependency-injection (v4.4.26 => v4.4.31)
  - Upgrading symfony/doctrine-bridge (v4.4.25 => v4.4.31)
  - Upgrading symfony/dom-crawler (v4.4.25 => v4.4.30)
  - Upgrading symfony/dotenv (v4.4.25 => v4.4.29)
  - Upgrading symfony/error-handler (v4.4.26 => v4.4.30)
  - Upgrading symfony/event-dispatcher (v4.4.25 => v4.4.30)
  - Upgrading symfony/expression-language (v4.4.25 => v4.4.30)
  - Upgrading symfony/filesystem (v4.4.26 => v4.4.27)
  - Upgrading symfony/finder (v4.4.25 => v4.4.30)
  - Upgrading symfony/flex (v1.13.3 => v1.16.3)
  - Upgrading symfony/form (v4.4.26 => v4.4.31)
  - Upgrading symfony/framework-bundle (v4.4.26 => v4.4.31)
  - Upgrading symfony/http-client-contracts (v1.1.10 => v2.4.0)
  - Upgrading symfony/http-foundation (v4.4.26 => v4.4.30)
  - Upgrading symfony/http-kernel (v4.4.26 => v4.4.32)
  - Upgrading symfony/inflector (v4.4.25 => v4.4.27)
  - Upgrading symfony/intl (v4.4.25 => v4.4.31)
  - Upgrading symfony/maker-bundle (v1.33.0 => v1.34.0)
  - Upgrading symfony/mime (v4.4.26 => v5.3.8)
  - Upgrading symfony/monolog-bridge (v4.4.26 => v4.4.27)
  - Upgrading symfony/options-resolver (v4.4.25 => v4.4.30)
  - Upgrading symfony/phpunit-bridge (v4.4.26 => v4.4.31)
  - Upgrading symfony/polyfill-mbstring (v1.23.0 => v1.23.1)
  - Upgrading symfony/polyfill-php80 (v1.23.0 => v1.23.1)
  - Upgrading symfony/process (v4.4.26 => v4.4.30)
  - Upgrading symfony/property-access (v4.4.25 => v4.4.30)
  - Upgrading symfony/proxy-manager-bridge (v4.4.25 => v4.4.27)
  - Upgrading symfony/routing (v4.4.25 => v4.4.30)
  - Upgrading symfony/security (v4.4.26 => v4.4.31)
  - Upgrading symfony/security-bundle (v4.4.26 => v4.4.27)
  - Upgrading symfony/serializer (v4.4.26 => v4.4.31)
  - Upgrading symfony/service-contracts (v1.1.9 => v2.4.0)
  - Upgrading symfony/stopwatch (v4.4.25 => v4.4.27)
  - Upgrading symfony/templating (v4.4.25 => v4.4.30)
  - Upgrading symfony/translation (v4.4.26 => v4.4.32)
  - Upgrading symfony/translation-contracts (v1.1.10 => v2.4.0)
  - Upgrading symfony/twig-bridge (v4.4.26 => v4.4.27)
  - Upgrading symfony/twig-bundle (v4.4.26 => v4.4.30)
  - Upgrading symfony/validator (v4.4.26 => v4.4.31)
  - Upgrading symfony/var-dumper (v4.4.26 => v4.4.31)
  - Upgrading symfony/var-exporter (v4.4.26 => v5.3.8)
  - Upgrading symfony/web-profiler-bundle (v4.4.26 => v4.4.31)
  - Upgrading symfony/web-server-bundle (v4.4.26 => v4.4.27)
  - Upgrading symfony/workflow (v4.4.25 => v4.4.27)
  - Upgrading symfony/yaml (v4.4.26 => v4.4.29)
  - Upgrading tecnickcom/tcpdf (6.4.1 => 6.4.2)
  - Upgrading theseer/tokenizer (1.1.3 => 1.2.1)
  - Upgrading twig/twig (v2.13.1 => v2.14.7)
  - Upgrading webmozart/assert (1.9.1 => 1.10.0)

nanasess committed:
cache:warmup が動作するところまで修正

nanasess committed:
TOPページの表示ができるよう修正

kurozumi committed:
会員パスワードの保存方法を変更

kurozumi committed:
テスト修正

kurozumi committed:
テスト修正

kurozumi committed:
テスト修正

kurozumi committed:
テスト修正

kurozumi committed:
テスト修正

nanasess committed:
Merge remote-tracking branch 'upstream/4.1' into improve/php8

nanasess committed:
[WIP] Symfony5 support

  - Removing bheller/images-generator (1.0.1)
  - Removing doctrine/doctrine-cache-bundle (1.4.0)
  - Removing doctrine/reflection (1.2.2)
  - Removing fzaninotto/faker (v1.9.2)
  - Removing jdorn/sql-formatter (v1.2.17)
  - Removing kylekatarnls/update-helper (1.2.1)
  - Removing laminas/laminas-eventmanager (3.4.0)
  - Removing laminas/laminas-zendframework-bridge (1.4.1)
  - Removing phpunit/php-token-stream (3.1.3)
  - Removing suncat/mobile-detect-bundle (v1.1.1)
  - Removing symfony/debug (v4.4.36)
  - Removing symfony/http-client-contracts (v2.5.0)
  - Removing symfony/mime (v5.4.2)
  - Removing symfony/security (v4.4.36)
  - Removing symfony/web-server-bundle (v4.4.27)
  - Upgrading codeception/codeception (4.1.28 => 4.1.31)
  - Upgrading codeception/phpunit-wrapper (7.8.2 => 9.0.7)
  - Upgrading codeception/stub (3.0.0 => 4.0.2)
  - Upgrading composer/composer (2.2.4 => 2.3.5)
  - Upgrading composer/package-versions-deprecated (1.11.99.4 => 1.11.99.5)
  - Upgrading composer/pcre (1.0.0 => 3.0.0)
  - Upgrading composer/semver (3.2.7 => 3.3.2)
  - Downgrading composer/xdebug-handler (2.0.4 => 2.0.2)
  - Upgrading dama/doctrine-test-bundle (v5.0.5 => v6.7.5)
  - Upgrading doctrine/common (2.13.3 => 3.2.2)
  - Upgrading doctrine/data-fixtures (1.5.1 => 1.5.2)
  - Upgrading doctrine/dbal (2.13.7 => 2.13.8)
  - Upgrading doctrine/doctrine-bundle (1.12.13 => 2.3.2)
  - Upgrading doctrine/doctrine-migrations-bundle (v1.3.2 => 3.2.2)
  - Upgrading doctrine/inflector (1.3.1 => 1.4.4)
  - Upgrading doctrine/instantiator (1.4.0 => 1.4.1)
  - Upgrading doctrine/lexer (1.2.2 => 1.2.3)
  - Upgrading doctrine/migrations (v1.8.1 => 3.4.2)
  - Upgrading doctrine/orm (2.7.5 => 2.8.5)
  - Upgrading doctrine/persistence (1.3.8 => 2.5.0)
  - Locking doctrine/sql-formatter (1.1.2)
  - Locking fakerphp/faker (v1.19.0)
  - Upgrading friendsofphp/proxy-manager-lts (v1.0.5 => v1.0.7)
  - Upgrading justinrainbow/json-schema (5.2.11 => 5.2.12)
  - Upgrading knplabs/knp-components (v1.3.10 => v3.5.0)
  - Upgrading knplabs/knp-paginator-bundle (v2.8.0 => v5.8.0)
  - Upgrading laminas/laminas-code (3.4.1 => 4.5.1)
  - Upgrading mobiledetect/mobiledetectlib (2.8.37 => 2.8.39)
  - Upgrading monolog/monolog (1.26.1 => 1.27.0)
  - Upgrading myclabs/deep-copy (1.10.2 => 1.11.0)
  - Upgrading nesbot/carbon (1.39.1 => 2.57.0)
  - Upgrading phar-io/manifest (1.0.3 => 2.0.3)
  - Upgrading phar-io/version (2.0.1 => 3.2.1)
  - Upgrading phpdocumentor/type-resolver (1.6.0 => 1.6.1)
  - Upgrading phpunit/php-code-coverage (6.1.4 => 9.2.15)
  - Upgrading phpunit/php-file-iterator (2.0.5 => 3.0.6)
  - Locking phpunit/php-invoker (3.1.1)
  - Upgrading phpunit/php-text-template (1.2.1 => 2.0.4)
  - Upgrading phpunit/php-timer (2.1.3 => 5.0.3)
  - Upgrading phpunit/phpunit (7.5.20 => 9.5.20)
  - Upgrading psr/container (1.1.1 => 1.1.2)
  - Locking psr/event-dispatcher (1.0.0)
  - Upgrading react/promise (v2.8.0 => v2.9.0)
  - Upgrading robthree/twofactorauth (1.8.1 => 1.8.2)
  - Locking sebastian/cli-parser (1.0.1)
  - Locking sebastian/code-unit (1.0.8)
  - Upgrading sebastian/code-unit-reverse-lookup (1.0.2 => 2.0.3)
  - Upgrading sebastian/comparator (3.0.3 => 4.0.6)
  - Locking sebastian/complexity (2.0.2)
  - Upgrading sebastian/diff (3.0.3 => 4.0.4)
  - Upgrading sebastian/environment (4.2.4 => 5.1.4)
  - Upgrading sebastian/exporter (3.1.4 => 4.0.4)
  - Upgrading sebastian/global-state (2.0.0 => 5.0.5)
  - Locking sebastian/lines-of-code (1.0.3)
  - Upgrading sebastian/object-enumerator (3.0.4 => 4.0.4)
  - Upgrading sebastian/object-reflector (1.1.2 => 2.0.4)
  - Upgrading sebastian/recursion-context (3.0.1 => 4.0.4)
  - Upgrading sebastian/resource-operations (2.0.2 => 3.0.3)
  - Locking sebastian/type (3.0.0)
  - Upgrading sebastian/version (2.0.1 => 3.0.2)
  - Upgrading seld/jsonlint (1.8.3 => 1.9.0)
  - Upgrading symfony/asset (v4.4.27 => v5.4.7)
  - Upgrading symfony/browser-kit (v4.4.27 => v5.4.3)
  - Upgrading symfony/cache (v4.4.36 => v5.4.7)
  - Upgrading symfony/cache-contracts (v2.5.0 => v2.5.1)
  - Upgrading symfony/config (v4.4.36 => v5.4.7)
  - Upgrading symfony/console (v4.4.36 => v5.4.7)
  - Upgrading symfony/css-selector (v4.4.27 => v5.4.3)
  - Upgrading symfony/debug-bundle (v4.4.36 => v5.4.3)
  - Upgrading symfony/dependency-injection (v4.4.36 => v5.4.7)
  - Upgrading symfony/deprecation-contracts (v2.5.0 => v2.5.1)
  - Upgrading symfony/doctrine-bridge (v4.4.36 => v5.4.7)
  - Upgrading symfony/dom-crawler (v4.4.36 => v5.4.6)
  - Upgrading symfony/dotenv (v4.4.36 => v5.4.5)
  - Upgrading symfony/error-handler (v4.4.36 => v5.4.7)
  - Upgrading symfony/event-dispatcher (v4.4.34 => v5.4.3)
  - Upgrading symfony/event-dispatcher-contracts (v1.1.11 => v2.5.1)
  - Upgrading symfony/expression-language (v4.4.34 => v5.4.7)
  - Upgrading symfony/filesystem (v4.4.27 => v5.4.7)
  - Upgrading symfony/finder (v4.4.36 => v5.4.3)
  - Upgrading symfony/flex (v1.17.6 => v1.18.5)
  - Upgrading symfony/form (v4.4.36 => v5.4.7)
  - Upgrading symfony/framework-bundle (v4.4.36 => v5.4.7)
  - Upgrading symfony/http-foundation (v4.4.36 => v5.4.6)
  - Upgrading symfony/http-kernel (v4.4.36 => v5.4.7)
  - Upgrading symfony/inflector (v4.4.34 => v5.4.3)
  - Upgrading symfony/intl (v4.4.36 => v5.4.5)
  - Upgrading symfony/maker-bundle (v1.36.4 => v1.38.0)
  - Upgrading symfony/monolog-bridge (v4.4.27 => v5.4.3)
  - Upgrading symfony/options-resolver (v4.4.30 => v5.4.3)
  - Locking symfony/password-hasher (v5.4.3)
  - Upgrading symfony/phpunit-bridge (v4.4.34 => v5.4.7)
  - Upgrading symfony/polyfill-ctype (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-iconv (v1.24.0 => v1.25.0)
  - Locking symfony/polyfill-intl-grapheme (v1.25.0)
  - Upgrading symfony/polyfill-intl-icu (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-intl-idn (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-intl-normalizer (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-mbstring (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-php72 (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-php73 (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-php80 (v1.24.0 => v1.25.0)
  - Upgrading symfony/polyfill-php81 (v1.24.0 => v1.25.0)
  - Upgrading symfony/process (v4.4.36 => v5.4.7)
  - Upgrading symfony/property-access (v4.4.36 => v5.4.7)
  - Locking symfony/property-info (v5.4.7)
  - Upgrading symfony/proxy-manager-bridge (v4.4.36 => v5.4.6)
  - Upgrading symfony/routing (v4.4.34 => v5.4.3)
  - Upgrading symfony/security-bundle (v4.4.36 => v5.4.5)
  - Locking symfony/security-core (v5.4.7)
  - Locking symfony/security-csrf (v5.4.3)
  - Locking symfony/security-guard (v5.4.3)
  - Locking symfony/security-http (v5.4.5)
  - Upgrading symfony/serializer (v4.4.36 => v5.4.7)
  - Upgrading symfony/service-contracts (v2.5.0 => v2.5.1)
  - Upgrading symfony/stopwatch (v4.4.27 => v5.4.5)
  - Locking symfony/string (v5.4.3)
  - Upgrading symfony/swiftmailer-bundle (v3.5.3 => v3.5.4)
  - Upgrading symfony/templating (v4.4.36 => v5.4.3)
  - Upgrading symfony/translation (v4.4.34 => v5.4.7)
  - Upgrading symfony/translation-contracts (v2.5.0 => v2.5.1)
  - Upgrading symfony/twig-bridge (v4.4.36 => v5.4.7)
  - Upgrading symfony/twig-bundle (v4.4.36 => v5.4.3)
  - Upgrading symfony/validator (v4.4.36 => v5.4.7)
  - Upgrading symfony/var-dumper (v4.4.36 => v5.4.6)
  - Upgrading symfony/var-exporter (v5.4.2 => v5.4.7)
  - Upgrading symfony/web-profiler-bundle (v4.4.31 => v5.4.6)
  - Upgrading symfony/workflow (v4.4.27 => v5.4.3)
  - Upgrading symfony/yaml (v4.4.36 => v5.4.3)
  - Upgrading twig/twig (v2.14.11 => v2.14.13)

nanasess committed:
suncat/mobile-detect-bundle は更新されていないため MobileDetect を直接利用する

nanasess committed:
FIXME env() の引数は文字列のみ

影響箇所を要修正

nanasess committed:
FIXME templating は廃止

影響箇所を要修正

nanasess committed:
FIXME Workflow の変更

影響箇所を要修正

nanasess committed:
FIXME ContainerAwareCommand の廃止

影響箇所を要修正

nanasess committed:
FIXME ControllerTrait の廃止

影響箇所を要修正

nanasess committed:
FIXME RoutingExtension に final が付与された

影響箇所を要修正

nanasess committed:
FIXME PasswordEncoderInterface に needsRehash が追加された

影響箇所を要修正

nanasess committed:
FIXME EventDispatcher::dispatch() に型パラメータが付与された

影響箇所を要修正

nanasess committed:
namespace の変更

nanasess committed:
Symfony\Bundle\FrameworkBundle\Controller\Controller の廃止

nanasess committed:
DataCollector::collect() の型パラメータ変更

nanasess committed:
config.platform.php: 7.4.0 で composer update を実行

nanasess committed:
doctrine/orm "^2.11" で composer update を実行

  - Removing composer/package-versions-deprecated (1.11.99.5)
  - Upgrading doctrine/doctrine-bundle (2.3.2 => 2.6.2)
  - Upgrading doctrine/orm (2.8.5 => 2.11.2)

nanasess committed:
doctrine/dbal "^3.3" で composer update

  - Upgrading doctrine/dbal (2.13.8 => 3.3.5)
  - Upgrading doctrine/migrations (3.4.2 => 3.5.0)

nanasess committed:
doctrine/cache "^2.0" で composer update

 - Upgrading doctrine/cache (1.12.1 => 2.1.1)

nanasess committed:
doctrine/inflector "^2.0 で composer update

- Upgrading doctrine/inflector (1.4.4 => 2.0.4)

nanasess committed:
friendsofphp/php-cs-fixer "^3.8" で composer update

  - Removing symfony/polyfill-php70 (v1.20.0)
  - Upgrading composer/xdebug-handler (2.0.2 => 3.0.3)
  - Upgrading friendsofphp/php-cs-fixer (v2.19.3 => v3.8.0)
  - Upgrading php-cs-fixer/diff (v1.3.1 => v2.0.2)

nanasess committed:
guzzlehttp/guzzle "^7.4" で composer update

  - Upgrading guzzlehttp/guzzle (6.5.5 => 7.4.2)
  - Upgrading guzzlehttp/psr7 (1.8.5 => 2.2.1)
  - Locking psr/http-client (1.0.1)
  - Locking psr/http-factory (1.0.1)

nanasess committed:
monolog/monolog "^2.5" で composer update

- Upgrading monolog/monolog (1.27.0 => 2.5.0)

nanasess committed:
Remove pimple/pimple

nanasess committed:
sensio/framework-extra-bundle "^6.2" で composer update

- Upgrading sensio/framework-extra-bundle (v5.6.1 => v6.2.6)

nanasess committed:
vlucas/phpdotenv "^5.4" で composer update

  - Locking graham-campbell/result-type (v1.0.4)
  - Locking phpoption/phpoption (1.8.1)
  - Upgrading vlucas/phpdotenv (v2.4.0 => v5.4.0)

nanasess committed:
トップページ表示できるところまで修正

nanasess commented:
PHP8.1 + Symfony5 でトップページ表示できるところまで修正しました
![スクリーンショット 2022-04-15 180342](https://user-images.githubusercontent.com/815715/163550926-3f61094b-ed4a-4dd4-aafa-7ce1a32d6f4c.png)

nanasess commented:
メモ) ControllerTrait が削除されたコミット
https://github.com/symfony/symfony/commit/30418cfaf3934a26698b7db220bdf909a4fbea2e

nanasess committed:
Merge remote-tracking branch 'upstream/4.1' into improve/symfony5

nanasess committed:
FIXME Twig3.3 に更新

- twig/extensions は非推奨のため削除
- Eccube\Twig\Extension\IntlExtension の修正が必要
- `Twig_*` クラスエイリアスを使用している箇所の修正が必要

nanasess committed:
PHPバージョンを 7.4-8.1 へ変更

nanasess committed:
PHPUnit9 support

- Apply composer recipes:install phpunit/phpunit --force -v
  - see https://symfony.com/doc/5.4/testing.html

nanasess committed:
symfony/dotenv で getenv が使用できなくなったため vlucas/phpdotenv を使用する

nanasess committed:
bheller\ImagesGenerator の廃止に伴い, --with-image オプションを削除

nanasess committed:
Doctrine\DBAL\Types の仕様変更対応

nanasess committed:
eccube:install コマンドを動作するよう修正

nanasess committed:
Twig_* の廃止対応

nanasess committed:
assertArraySubset 廃止に伴う修正

see https://github.com/sebastianbergmann/phpunit/issues/3494

nanasess committed:
Intl component の仕様変更に伴う修正

nanasess committed:
EmailValidator の仕様変更に伴う修正

- `checkMX` 及び `checkHost` は削除された
- 引数の変更

nanasess committed:
Doctrine Persistence の仕様変更対応

nanasess committed:
EmailValidator の仕様変更に伴う修正

- 引数の変更

nanasess committed:
DateType, DateTimeType の仕様変更に伴う修正

- `html5` オプションが true の場合は format オプションは指定不可

nanasess committed:
EmailValidator の仕様変更に伴う修正

- 引数の変更

nanasess committed:
Doctrine\DBAL\Driver::getName() は廃止

nanasess committed:
Symfony Event の仕様変更対応

nanasess committed:
Doctrine Persistence の仕様変更対応

nanasess committed:
ControllerTrait 廃止に伴う修正

nanasess committed:
Twig-Extensions 廃止に伴う修正

localizeddate を format_date に修正

nanasess committed:
assertContains の仕様変更に伴う修正

nanasess committed:
SameSiteCookie テストケースの修正

- vlucas/phpdotenv のバージョンアップ対応

nanasess committed:
WebTestCase::createClient() の仕様変更対応

Fix `LogicException: Booting the kernel before calling "Symfony\Bundle\FrameworkBundle\Test\WebTestCase::createClient()" is not supported, the kernel should only be booted once.`

chihiro-adachi committed:
Symfony\Processの仕様変更対応

chihiro-adachi committed:
MetadataDriverの仕様変更対応

chihiro-adachi commented:
メモ) プラグインのルーティングが読み込まれない

~configureRoutesがprivateメソッドになっている~

- https://github.com/EC-CUBE/ec-cube/blob/4.1/src/Eccube/Kernel.php#L169
	- https://github.com/symfony/framework-bundle/blob/4.4/Kernel/MicroKernelTrait.php#L33
	- https://github.com/symfony/framework-bundle/blob/5.4/Kernel/MicroKernelTrait.php#L74

~※ configureContainerも同様。services.yamlが読めなくなってるかも~

メモ）プラグインのナビ拡張が読み込まれない

~原因は不明~

https://github.com/nanasess/ec-cube/pull/54 で修正
DBの接続判定に失敗し、有効なプラグインのリストを取得できなかった

nanasess committed:
Merge pull request #51 from chihiro-adachi/improve/symfony54

Improve/symfony54

chihiro-adachi committed:
Fix "implode(): Argument #2 ($array) must be of type ?array, string given"

chihiro-adachi committed:
Fix 'Attempted to call an undefined method named "fetchColumn" of class "Doctrine\DBAL\Result" / "Doctrine\DBAL\Connection"'

chihiro-adachi committed:
Symfony\Processの仕様変更対応

nanasess committed:
Merge pull request #52 from chihiro-adachi/improve/symfony5

Improve/symfony5

chihiro-adachi committed:
DBへの接続判定を修正

nanasess committed:
PostgreSQL と MySQL のセットアップに失敗していたのを修正

- phpdotenv のバージョンアップに伴い挙動が変わった模様

nanasess committed:
Merge pull request #54 from chihiro-adachi/improve/symfony5

DBへの接続判定を修正

nanasess committed:
UploadedFile::__construct() の仕様変更対応

nanasess committed:
Merge remote-tracking branch 'origin/improve/symfony5' into improve/symfony5

nanasess committed:
WebTestCase::createClient() の仕様変更対応

nanasess committed:
ContainerAwareCommand が廃止されたのに伴う対応

- `An option named "em" already exists.` と出力されていたのを修正

nanasess committed:
Merge remote-tracking branch 'upstream/4.1' into improve/symfony5

nanasess committed:
ChoiceList の仕様変更対応

- Fix `TypeError: Argument 1 passed to Symfony\Bridge\Doctrine\Form\ChoiceList\IdReader::getIdValue() must be an object or null, int given`

nanasess committed:
XmlEncoder の仕様変更対応

Fix `TypeError: Argument 1 passed to Symfony\Component\Serializer\Encoder\XmlEncoder::__construct() must be of the type array, string given`

nanasess committed:
Merge branch 'fix-save-customer-password' into improve/symfony5

nanasess committed:
Codeception を動作するよう修正

nanasess committed:
OrderStateMachine の仕様変更対応

nanasess committed:
OrderStateMachine の仕様変更対応

nanasess committed:
SwiftMailer が非推奨になったため SymfonyMailer に変更する

nanasess committed:
Merge remote-tracking branch 'origin/improve/symfony5' into improve/symfony5

chihiro-adachi committed:
mailer.yamlを追加
https://symfony.com/doc/5.4/mailer.html#transport-setup

nanasess committed:
Merge pull request #55 from chihiro-adachi/improve/symfony5

mailer.yamlを追加

nanasess committed:
SwiftMailer が非推奨になったため SymfonyMailer に変更する

nanasess committed:
Merge remote-tracking branch 'origin/improve/symfony5' into improve/symfony5

nanasess committed:
SwiftMailer が非推奨になったため SymfonyMailer に変更する

nanasess committed:
Symfony workflow の仕様変更対応

以下のエラー修正
> The value "single_state" is not allowed for path "framework.workflows.workflows.order.marking_store.type". Permissible values: "method"

nanasess committed:
Remove swiftmailer.yaml

nanasess committed:
Fix MAILER_DSN

nanasess committed:
MetadataDriver の仕様変更対応

nanasess commented:
~~`Symfony\Component\HttpFoundation\File\UploadedFile` を使用したテストが動かず。 `FormEvents::PRE_SUBMIT` で `$event->getData()` をコールすると、 `File` オブジェクトが返ってくるはずだがテキスト文字列になってしまいアップロードエラーになってしまう。
`Symfony\Component\HttpFoundation\File\UploadedFile`  を使用したテスト方法は、 Symfony4.4 でドキュメントから削除されている
https://github.com/symfony/symfony-docs/pull/15304~~

### 追記

https://github.com/EC-CUBE/ec-cube/pull/5353/commits/312f52655f82f56949474c431690884dc6be16dc にて修正

chihiro-adachi commented:
@nanasess 
手元の環境では、PaymentControllerTestはすんなり通ったのですが他のテストでしょうか？
php8.0, phpunit 9.5.20です。

```
$ bin/phpunit tests/Eccube/Tests/Web/Admin/Setting/Shop/PaymentControllerTest.php
PHPUnit 9.5.20 #StandWithUkraine

Testing Eccube\Tests\Web\Admin\Setting\Shop\PaymentControllerTest
..............                                                    14 / 14 (100%)

Time: 00:01.399, Memory: 68.50 MB


```

nanasess commented:
@chihiro-adachi `Eccube\Tests\Web\Admin\Order\CsvImportControllerTest` が通らないですね。 `CsvImportType` が問題なのかも 🤔

nanasess commented:
@chihiro-adachi この修正で通りました。ありがとうございます！

```diff
diff --git a/tests/Eccube/Tests/Web/Admin/Order/CsvImportControllerTest.php b/tests/Eccube/Tests/Web/Admin/Order/CsvImportControllerTest.php
index 5794888a70..112b6d89eb 100644
--- a/tests/Eccube/Tests/Web/Admin/Order/CsvImportControllerTest.php
+++ b/tests/Eccube/Tests/Web/Admin/Order/CsvImportControllerTest.php
@@ -176,7 +176,7 @@ class CsvImportControllerTest extends AbstractAdminWebTestCase
                     'import_file' => $file,
                 ],
             ],
-            ['import_file' => $file]
+            ['admin_csv_import' => ['import_file' => $file]]
         );

         $this->assertRegexp(
```

nanasess committed:
ファイルアップロードの仕様変更対応

nanasess committed:
Twig の仕様変更対応

以下を修正

> ArgumentCountError: Too few arguments to function Twig\Environment::loadTemplate()

nanasess committed:
Twig-Extensions 廃止に伴う修正

b3ffc8c7ed9a01d42ca1f000b761414e9f8b9777 の残修正

nanasess committed:
ファイルアップロードの仕様変更対応

nanasess committed:
ContainerInterface の修正

以下のエラーを修正
> Error: Call to undefined method Symfony\Component\DependencyInjection\Argument\ServiceLocator::getParameter()

nanasess committed:
assertNotContains の仕様変更に伴う修正

nanasess committed:
Doctrine DBAL の仕様変更対応

nanasess committed:
Revert "FIXME RoutingExtension に final が付与された"

This reverts commit 744714f35ec042565bb4a3a2ad8ed2ffe307e03e.

nanasess committed:
RoutingExtension の仕様変更対応

nanasess committed:
Fix sql syntax error

nanasess committed:
EventDispacher の仕様変更

nanasess committed:
配列の項目に null を渡すと有効な値ではありませんというエラーになる

nanasess committed:
Service Container の仕様変更

Controller の $this->container は `Psr\Container\ContainerInterface`
をインジェクションするようになり、 Service Locator の設定が必要となった

https://symfony.com/doc/5.4/service_container/service_subscribers_locators.html

従来のように、 public service を取得したい場合は、必要に応じて
`Symfony\Component\DependencyInjection\ContainerInterface` をインジェクションする必要がある

nanasess committed:
SwiftMailer が非推奨になったため SymfonyMailer に変更する

nanasess committed:
Email の Assertion を修正

nanasess committed:
DateType の仕様変更対応

`Assert\DateTime()` でのバリデーションはテキスト文字列のみを許容する.
DateType は `DateTime` 型の値が渡ってしまうため常にエラーとなってしまう

nanasess committed:
ユニットテストの軽微なエラー修正

nanasess committed:
Twig "^2.15" に変更

Twig3 で `base_template_class` オプションが削除されたのに伴い、
EC-CUBE の TemplateEvent が動作しなくなってしまう。
https://github.com/twigphp/Twig/blob/a1f8d3f91a0d28c258e81607e6711387a46f1705/CHANGELOG#L142

代替方法が見つからないこと、 Twig2 はメンテナンスが継続していることから、 Twig 2.15.x を使用する

nanasess committed:
Fix deprecated to PHPUnit

nanasess committed:
SYMFONY_DEPRECATIONS_HELPER=weak

nanasess committed:
Apply composer update composer/composer

- Upgrading composer/composer (2.2.12 => 2.3.5)
- 以下の不具合修正のため
  > TypeError: Argument 3 passed to Composer\Pcre\Preg::isMatch() must be of the type array or null, bool given

nanasess committed:
PHPUnit9 対応に関する修正

`@expectedException` は廃止されたため、`$this->expectException()` を使用する

nanasess committed:
テスト用のプラグインの 4.2 対応

nanasess committed:
Fix assertContains to assertStringContainsString

nanasess committed:
PostgreSQL でテストが落ちていたのを修正

https://github.com/EC-CUBE/ec-cube/runs/6550254592?check_suite_focus=true#step:9:157

nanasess committed:
Symfony Console の仕様変更に伴う修正

nanasess committed:
Fix `Syntax error or access violation: 1305 SAVEPOINT DOCTRINE2_SAVEPOINT_3 does not exist`

see https://github.com/dmaicher/doctrine-test-bundle/issues/58#issuecomment-391081408

nanasess committed:
Bundle プラグインの依存バージョンを修正

Symfony5.4 に対応するためには trikoder/oauth2-bundle "v3.x-dev@dev" に依存する必要がある

nanasess committed:
Bundle テスト用プラグインに依存関係追加

"ajgarlag/psr-http-message-bundle": "^1.2"

nanasess committed:
Fix no active transaction

PHP8 with MySQL で no active transaction になるのを修正.

chihiro-adachi committed:
インストーラでMAILER_DNSの未定義エラーが発生するのを修正

chihiro-adachi committed:
マイグレーションファイルのパスを定義

chihiro-adachi committed:
fetchColumn -> fetchOne

nanasess committed:
Bundle テスト用プラグインで Bundle が二重定義されてしまう問題の回避

nanasess committed:
Fix MappingException

nanasess committed:
`env()` の仕様変更

- boolean は文字列の '0', '1' に変更する

chihiro-adachi committed:
PluginManager::migrationを動作するように修正

nanasess committed:
Merge pull request #56 from chihiro-adachi/improve/symfony5

マイグレーションを動作するように修正

nanasess committed:
PHP8.1 対応

- `#[\ReturnTypeWillChange]` アノテーションを付与
- デフォルト値のある引数はすべて右端にする必要がある

nanasess committed:
PluginManagerTest に @group plugin-service を付与

`Syntax error or access violation: 1305 SAVEPOINT DOCTRINE2_SAVEPOINT_2 does not exist` の対策

nanasess committed:
Fix MAILER_URL to MAILER_DSN

nanasess committed:
PHPUnit9 対応

captbaritone/mailcatcher-codeception-module "^2.4" へ更新

nanasess committed:
Merge branch '4.1' into improve/symfony5

nanasess committed:
Fix deprecated

- RequestStack::getMasterRequest() to RequestStack::getMainRequest()
- KernelEvent::isMasterRequest() to KernelEvent::isMainRequest()

nanasess committed:
ファイルダウンロード時にシステムエラーが発生するのを防止

以下の Warning が発生してファイルダウンロードに失敗する
`Warning: Cannot modify header information - headers already sent by ...`

nanasess committed:
Webインストーラを動作するよう修正

nanasess committed:
ChoiceLoader の仕様変更対応

- 商品一覧の初期ソート順がID降順になってしまうのを修正

nanasess committed:
Bump version to PHPStan 1.7

- PHPStan のエラー修正

nanasess committed:
非推奨のメソッド削除

以下を削除
- 現状のロジックに影響のないメソッド
- スキーマ変更を伴わないもの

nanasess committed:
非推奨メソッドを使用していた箇所の修正

`php_*` を修正

nanasess committed:
Codeception のエラー修正

nanasess committed:
Codeception のエラー修正

nanasess committed:
FormType のドロップダウンリストが正常に選択されないのを修正

リクエストをハンドリングするFormTypeと表示するFormTypeがわかれていると、
リクエストとは違う値が選択されてしまう.

ProductListMaxType, ProductListOrderBy は SearchProductType で一元管理
する必要がある

nanasess committed:
Merge remote-tracking branch 'origin/improve/symfony5' into improve/symfony5

nanasess committed:
Codeception のエラー修正

nanasess committed:
非推奨メソッドを使用していたテストを修正

nanasess committed:
デバッグツールバーが動作しないのを修正

chihiro-adachi commented:
@nanasess 
こちらの修正確認しました。
https://github.com/EC-CUBE/ec-cube/pull/5353/commits/759d4060fc3fcb8f6a1f81f5930f11aaa9383662

内容問題ないかと思います。
ただ、パラメータの値が従来と変更になるので、仕様変更としてマイグレーションガイド等に記載必要そうですね。

4.0/4.1
/products/list?mode=&category_id=&name=&pageno=1&disp_number=[0, 1, 2]&orderby=[0, 1, 2]

4.2
/products/list?mode=&category_id=&name=&pageno=1&disp_number=[20, 40, 60]&orderby=[1, 2, 3]

chihiro-adachi committed:
proxyの読み込みタイミングを調整
Compile Error: Cannot declare class Eccube\Entity\Customer, because the name is already in use

chihiro-adachi commented:
@nanasess 
プラグインインストール時に、proxyの読み込みでエラーとなるようで、prしています。
https://github.com/nanasess/ec-cube/pull/57

nanasess committed:
Merge pull request #57 from chihiro-adachi/improve/symfony5

proxyの読み込みタイミングを調整

nanasess commented:
@chihiro-adachi マージしました。ありがとうございます！

nanasess committed:
RFC違反のメールを送信できるよう修正

パラメータ `eccube_rfc_email_check` が false の場合, RFC違反のメールア
ドレスはメール送信する際に local part を "" で囲む.
true の場合は囲まない.

nanasess committed:
Merge remote-tracking branch 'origin/improve/symfony5' into improve/symfony5

nanasess commented:
ユニットテスト、 E2Eテスト共にオールグリーンになりましたので WIP はずしました

chihiro-adachi commented:
ありがとうございます。4.2ブランチへマージしました。
